portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,39 +34,70 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+ /** 
+  * This class represents CommentDescription.
+  */
+  public class UserComment {
+      private String emailId;
+      private String comment;
+      UserComment(String emailId,String comment) {
+          this.emailId = emailId;
+          this.comment = comment;
+      }
+  }
   private static final String INVALID_COMMENT_RESPONSE_STRING = "Please enter a valid comment";
   private static final String ENTITY_NAME = "Comment";
   private static final String ENTITY_NAME_TIMESTAMP = "timestamp";
+  private static final String ENTITY_NAME_EMAILID = "EmailId";
 
+  /**
+  * Handler for server side POST requests.
+  */
   @Override
   public void doPost(HttpServletRequest request,HttpServletResponse response) throws IOException {
-      String comment = request.getParameter(ENTITY_NAME);
-      long timestamp = System.currentTimeMillis();
-      if(comment.length()==0) {
-          response.setContentType("text/html;");
-          response.getWriter().println(INVALID_COMMENT_RESPONSE_STRING);
-          return;
-      }
-      Entity entity = new Entity(ENTITY_NAME);
-      entity.setProperty(ENTITY_NAME,comment);
-      entity.setProperty(ENTITY_NAME_TIMESTAMP,timestamp);
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      datastore.put(entity);
+    UserService userService = UserServiceFactory.getUserService();
+
+    if(!userService.isUserLoggedIn()) {
       response.sendRedirect("/index.html");
+      return;
+    }
+
+    String comment = request.getParameter(ENTITY_NAME);
+    long timestamp = System.currentTimeMillis();
+    if(comment.length()==0) {
```

Comment:
Space before and after equality operator.

```suggestion
    if(comment.empty()) {
```